pkg/armhelpers/deployer.go

```diff
@@ -0,0 +1,59 @@
+package armhelpers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/prometheus/common/log"
+)
+
+type AzureTemplateDeployer struct{}
+
+func (az *AzureClient) Validate(resourceGroup, name string, d resources.Deployment) (resources.DeploymentValidateResult, error) {
+	return az.DeploymentsClient.Validate(resourceGroup, name, d)
+}
+
+func (az *AzureClient) CreateOrUpdate(resourceGroup, name string, d resources.Deployment, cancel chan struct{}) (<-chan resources.DeploymentExtended, <-chan error) {
+	return az.DeploymentsClient.CreateOrUpdate(resourceGroup, name, d, cancel)
+}
+
+func (az *AzureClient) Delete(resourceGroup, name string, cancel chan struct{}) (<-chan autorest.Response, <-chan error) {
+	return az.DeploymentsClient.Delete(resourceGroup, name, cancel)
+}
+
+func (az *AzureClient) Get(resourceGroup, name string) (resources.DeploymentExtended, error) {
+	return az.DeploymentsClient.Get(resourceGroup, name)
+}
+
+func DeployTemplate(deployer TemplateDeployer, resourceGroupName, deploymentName string, template map[string]interface{}, parameters map[string]interface{}) (response *resources.DeploymentExtended, err error) {
```

Comment:
The point is that it's a utility function that works with anything that implements TemplateDeployer. The struct is simply unnecessary. But I don't understand this question: `shouldn't CreateOrUpdate be associated with AzureTemplateDeployer?` I think the question just goes away when I remove the AzureTemplateDeployer struct that should've never been in the PR.